Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1832,7 +1818,7 @@ private InvalidBlockException ValidateNextBlock(Block<T> nextBlock)
             _rwlock.EnterUpgradeableReadLock();
             try
             {
-                if (offset is null && Tip is null)
+                if (offset is null && Count == 0)
```

Comment:
How about removing this `if` block at all?

Reply:
I agree the logic looks little weird. It seems to be more of a safety measure than a feature. The only possible failure is when trying to access the states before `BlockChain<T>` construction is complete. I'll try removing it and see if it breaks anything. 🙃

Reply:
Done. See c99cde40e6c645466bf15cf78f4d7109388c865e.

Reply:
Arg... this has been reverted back. Since `genesis` is appended just like any other block via `Append()`, state look-up is performed while `Tip` is not accessible. Due to this, many tests break in Lib9c. 😞